webapp/src/ts/components/actionbar/actionbar.component.html

```diff
@@ -148,79 +166,102 @@
+              <li *ngFor="let form of actionBar.right.relevantForms; trackBy: trackByForms" [attr.id]="form.id">
+                <a (click)="actionBar.right.openContactMutedModal($event, form)">
                   <span [innerHTML]="form.icon | resourceIcon"></span>
-                  <span class="content">{{form.title}}</span>
+                  <span class="content">{{ form.title }}</span>
                 </a>
               </li>
             </ul>
           </span>
 
-          <span *ngFor="let group of actionBar?.right?.childTypes" mmAuth="{{group.permission}}">
-
-            <a class="mm-icon mm-icon-inverse mm-icon-caption dropdown-toggle" data-toggle="dropdown" *ngIf="group.types.length > 1">
-              <span class="fa-stack">
-                <i class="fa {{group.menu_icon}}"></i>
-                <i class="fa fa-plus fa-stack-1x"></i>
-              </span>
-              <p translate>{{group.menu_key}}</p>
-            </a>
-            <ul class="dropdown-menu mm-dropdown-menu with-icon" *ngIf="group.types.length > 1">
-            TODO order group.types by 'id' !!
-              <li *ngFor="let type of group.types; trackBy: type.id">
-                <a ui-sref="contacts.addChild({ type: type.id, parent_id: actionBarCtrl.selectedContactDoc._id })">
-                  <span [innerHTML]="type.icon | resourceIcon"></span>
-                  <span class="content" translate>{{type.create_key}}</span>
-                </a>
-              </li>
-            </ul>
+          <span *ngFor="let group of actionBar?.right?.childTypes" mmAuth="{{ group.permission }}">
+            <ng-container *ngIf="group.types?.length > 1">
+              <a class="mm-icon mm-icon-inverse mm-icon-caption dropdown-toggle" data-toggle="dropdown">
+                <span class="fa-stack">
+                  <i class="fa {{ group.menu_icon }}"></i>
+                  <i class="fa fa-plus fa-stack-1x"></i>
+                </span>
+                <p translate>{{ group.menu_key }}</p>
+              </a>
+              <ul class="dropdown-menu mm-dropdown-menu with-icon">
+                <li *ngFor="let type of group.types; trackBy: trackById">
+                  <a (click)="navigateToContactAdd(type.id, selectedContactDoc._id)">
+                    <span [innerHTML]="type.icon | resourceIcon"></span>
+                    <span class="content" translate>{{ type.create_key }}</span>
+                  </a>
+                </li>
+              </ul>
+            </ng-container>
 
-            <a class="mm-icon mm-icon-inverse mm-icon-caption" *ngIf="group.types.length === 1" ui-sref="contacts.addChild({ type: group.types[0].id, parent_id: selectedContactDoc._id })">
+            <a *ngIf="group.types?.length === 1"
+               class="mm-icon mm-icon-inverse mm-icon-caption"
+               (click)="navigateToContactAdd(group.types[0]?.id, selectedContactDoc._id)"
```

Comment:
Same as above, can this be a link instead of a click event?

Reply:
Router link added